client/my-sites/checkout/src/components/wp-contact-form.tsx

```diff
@@ -56,26 +75,31 @@ export default function WPContactForm( {
 	isLoggedOutCart: boolean;
 	setShouldShowContactDetailsValidationErrors: ( allowed: boolean ) => void;
 } ) {
+	const translate = useTranslate();
 	const contactInfo = useSelect( ( select ) => select( CHECKOUT_STORE ).getContactInfo(), [] );
 	const { formStatus } = useFormStatus();
 	const isStepActive = useIsStepActive();
 	const isDisabled = ! isStepActive || formStatus !== FormStatus.READY;
 
-	usePrefillCheckoutContactForm( {
+	const hasCompleted = usePrefillCheckoutContactForm( {
 		setShouldShowContactDetailsValidationErrors,
 		isLoggedOut: isLoggedOutCart,
 	} );
 
 	return (
-		<BillingFormFields>
-			<ContactDetailsContainer
-				contactDetailsType={ contactDetailsType }
-				contactInfo={ contactInfo }
-				countriesList={ countriesList }
-				shouldShowContactDetailsValidationErrors={ shouldShowContactDetailsValidationErrors }
-				isDisabled={ isDisabled }
-				isLoggedOutCart={ isLoggedOutCart }
-			/>
+		<BillingFormFields isLoading={ ! hasCompleted }>
+			{ hasCompleted ? (
+				<ContactDetailsContainer
+					contactDetailsType={ contactDetailsType }
+					contactInfo={ contactInfo }
+					countriesList={ countriesList }
+					shouldShowContactDetailsValidationErrors={ shouldShowContactDetailsValidationErrors }
+					isDisabled={ isDisabled }
+					isLoggedOutCart={ isLoggedOutCart }
+				/>
+			) : (
+				<LoadingText>{ translate( 'Retrieving contact details…' ) }</LoadingText>
```

Comment:
Weird. I don't see this in GlotPress. Not sure why string freeze hasn't picked this up. @Automattic/i18n any ideas?

Reply:
It is now visible in GlotPress

Reply:
@Automattic/i18n 

I believe the same issue happened in this PR: https://github.com/Automattic/wp-calypso/pull/99325

Reply:
Yep, I think you just need to push changes/rebase, then also readd the string freeze label. It should refresh it and get the originals back into GlotPress. The problem is they're marked as obsolete after 2 weeks I think, hence also the case in https://github.com/Automattic/wp-calypso/pull/99325

I think we'll be updating some of the messaging in upcoming refactors

Reply:
@chriskmnds Why would it say the translations are complete?

<img width="874" alt="Screenshot 2025-03-06 at 2 48 15 PM" src="https://github.com/user-attachments/assets/c2187dbd-77af-4614-a70e-273d2f030475" />

Reply:
They are complete now. I checked in GP and it's been translated. The problem is it gets marked as obsolete if in a PR for +2 weeks. Refreshing and readding the labels pulls it back

Reply:
Ah, if the **PR** is +2 weeks old. Understood. I would expect the system to care about when I added the String Freeze label. The age of the PR is irrelevant.

Reply:
> Ah, if the PR is +2 weeks old.

I'm not sure if that's the case. I think it's as of when the string freeze PR's been translated or added for translation, like you expect. Not sure what the problem was with https://github.com/Automattic/wp-calypso/pull/99325 (I didn't when the label was added before commenting) :-)